src/AWSCredentials.jl

```diff
@@ -461,15 +461,18 @@ function credentials_from_webtoken()
 
     has_all_keys = 
         haskey(ENV, token_role_arn) &&
-        haskey(ENV, token_role_session) &&
         haskey(ENV, token_web_identity)
 
     if !has_all_keys
         return nothing
     end
 
     role_arn = ENV[token_role_arn]
-    role_session = ENV[token_role_session]
+    role_session = get!(ENV, token_role_session) do
```

Comment:
Do we need to permanently modify the `ENV`?
```suggestion
    role_session = get(ENV, token_role_session) do
```

Reply:
> Do we need to permanently modify the `ENV`?

No need really, but also once this has run you don't have a trace of what the generated session name ended up being outside of logs. What's worst, having no trace or setting an ENV var? Can't think of a third way.

I guess it's best not to pollute ENV.